implementations/rust/ockam/ockam_transport_tcp/src/trust_options.rs

```diff
@@ -0,0 +1,103 @@
+use ockam_core::compat::sync::Arc;
+use ockam_core::sessions::{SessionId, SessionOutgoingAccessControlBuilder, Sessions};
+use ockam_core::{IncomingAccessControl, LocalOnwardOnly, LocalSourceOnly, OutgoingAccessControl};
+
+pub(crate) struct TcpConnectionAccessControl {
+    pub fresh_session_id: Option<SessionId>,
+    pub sender_incoming_access_control: Arc<dyn IncomingAccessControl>,
+    pub receiver_outgoing_access_control: Arc<dyn OutgoingAccessControl>,
+}
+
+/// Trust Options for a TCP connection
+#[derive(Clone, Default, Debug)]
+pub struct TcpConnectionTrustOptions {
+    pub(crate) session: Option<(Sessions, SessionId)>,
+}
+
+impl TcpConnectionTrustOptions {
+    /// Constructor
+    pub fn new() -> Self {
+        Self { session: None }
+    }
+
+    /// Set session for this connection, in this case messages from that connection
+    /// will be only allowed to go to the [`Address`]es with the same [`SessionId`].
+    /// Information of [`Address`]' [`SessionId`] is stored in [`Sessions`] struct.
+    ///
+    /// Also this [`SessionId`] will be added to [`LocalInfo`] of the messages from that
+    /// connection
+    pub fn with_session(mut self, sessions: &Sessions, session_id: &SessionId) -> Self {
```

Comment:
I didn't get your suggestion. The idea is to have globally shared `Sessions` storage, so you only need its id to access a specific session. At this point global state is inside a shared `Sessions`, later, that could be embedded in `ockam_node` so that everyone has access to it using `Context`.

Reply:
My suggestion is to _not_ put `Sessions` inside the trust options which concern only one connection/listener and pass it later where it is needed. We will have to do that eventually anyway as you write when we access it from a `Context`

Reply:
But it's already meant to be shared (and cloning actually gives you an object with access to the same storage)